src/pairing_scheduler/core.clj

```diff
@@ -107,13 +107,26 @@
                         [guest-id (individual-score guest-id context)]))
                  (into {}))))
 
+;; if a user has no preferred timeslots,
+;; update all available timeslots to preferred.
+(defn update-preferred
+  [{:keys [schedule availabilities] :as context}]
+  (->> availabilities
+       (map (fn [[key value]]
+          (if (= 0 (count (filter (fn [x] (= (x 2) :preferred)) value)))
+            [key (set (map (fn [x] [(x 0) (x 1) :preferred]) value))]
+            [key value])))
+       (into {})
+       (assoc context :availabilities)))
```

Comment:
Instead, do `(update context :availabilities (fn [availabilities]  .... ))`

Reply:
couldn't get (update context :availabilities (fn [availabilities] .... )) to work.